zeebe/broker/src/main/java/io/camunda/zeebe/broker/engine/impl/BoundedCommandCache.java

```diff
@@ -64,8 +64,32 @@ public BoundedCommandCache(final int capacity, final IntConsumer sizeReporter) {
     sizeReporter.accept(0);
   }
 
-  public void add(final LongHashSet keys) {
-    LockUtil.withLock(lock, () -> lockedAdd(keys));
+  public void add(final long key) {
+    LockUtil.withLock(
+        lock,
+        () -> {
+          final int evictionCount = cache.size() + 1 - capacity;
+          if (evictionCount > 0) {
+            evict(evictionCount);
+          }
+
+          cache.add(key);
+          sizeReporter.accept(cache.size());
+        });
```

Comment:
:upside_down_face: I find keeping the synchronized methods in private method makes it easier to separate business logic from dealing with concurrency, and thus more readable. Since we don't make use of things like the Checker framework to check for `@GuardedBy` annotations, it's purely for humans - but I would recommend it in general. No super strong argument though.

Reply:
Done with ae875a55005198a05823f0daf499d7a6a60dfde6